cmd/soroban-cli/src/cli.rs

```diff
@@ -72,6 +76,13 @@ pub async fn main() {
 
     if let Err(e) = root.run().await {
         eprintln!("error: {e}");
-        std::process::exit(1);
+        exit(1);
     }
+
+    print_upgrade_prompt(root.global_args.quiet);
```

Comment:
a3d935966aa0aeee3b5fa1b234290a680a50c3c8

Hi @leighmcculloch, currently, I have still modified it to run in a separate thread. But I still doubt whether this is better than the previous design.

The previous version was intentional by me, because I wanted this check to always be executed, so that users can see this prompt when encountering issues due to using a lower version of the CLI. Also, I think it should only make one HTTP request per day and have a timeout configuration of 30 seconds, which should be an acceptable resource consumption. Additionally, users can disable it by configuring environment variables.

Reply:
> I wanted this check to always be executed

I understand the motivation here, but the upgrade check is a convenience and secondary feature. I think it's important that it nevers reduces the experience of any commands primary functionality. It's important that folks don't see 30 second delays (the timeout period) if they're on spotty wifi, or have slow internet. Especially when running commands that otherwise don't make any connection to the outside world.

I understand this will mean some fast commands won't see the message until a slower command has picked it up, then fast commands will display the warning. I think this is an appropriate trade off.